src/syntax/syntax.rkt

```diff
@@ -60,44 +60,136 @@
 ;; TODO: the costs below seem likely to be incorrect, and also do we still need them?
 (define-table operators
   [args  (listof (or/c '* natural-number/c))]
-  [bf    (unconstrained-argument-number-> (or/c bigfloat? boolean?) (or/c bigfloat? boolean?))]
-  [fl    (unconstrained-argument-number-> (or/c flonum? boolean?) (or/c flonum? boolean?))]
+  [bf    (unconstrained-argument-number-> (or/c bigfloat? boolean? pair?) (or/c bigfloat? boolean? pair?))]
```

Comment:
Here too, changing `pair?` to `bigcomplex?` would make things more readable and also check for more errors.